File: path-sum/path-sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: TreeNode, targetSum: int) -> bool:
        '''
        Determines if a route-to-leaf path contains the target sum
        Returns True if a route exists, false otherwise
        '''
        if not root: #Root contains nothing
            return False
        if targetSum == root.val and not root.left and not root.right: #Check if targetSum reached
            return True
        current = targetSum - root.val #Update target val
        return (self.hasPathSum(root.left, current) or self.hasPathSum(root.right, current)) #Recurse for result

File: path-sum/path-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution 
{
    public boolean hasPathSum(TreeNode root, int targetSum) 
    {
        if (root == null)
            return false;
        return DFS(root, targetSum);
    }
    public boolean DFS(TreeNode root, int target)
    {
        if (root == null)
            return false;
        if (target - root.val == 0)
            if (root.left == null && root.right == null)
                return true;
        return DFS(root.left, target-root.val) || DFS(root.right, target-root.val);
    }
}